koiki/create_delivery.py

```diff
@@ -1,29 +1,61 @@
-from koiki.sender import Sender
-from koiki.recipient import Recipient
-from koiki.order import Order
+from koiki.models import Sender, Recipient, Shipment
+from koiki.woocommerce.models import LineItem, Shipping, Billing
 
 
 class CreateDelivery():
     LABEL_FORMAT = 'PDF'
     RESOURCE_PATH = '/altaEnvios'
 
     def __init__(self, order):
-        self.order = Order(order)
-        self.recipient = Recipient(order)
-        self.sender = Sender()
+        self.order = order
+        self.by_vendor = self._by_vendor(order['line_items'])
+        self.shipping = Shipping(order['shipping'])
+        self.billing = Billing(order['billing'])
 
     def body(self):
         return {
             'formatoEtiqueta': self.LABEL_FORMAT,
-            'envios': [self._delivery()]
+            'envios': self._deliveries()
         }
 
     def url(self):
         return self.RESOURCE_PATH
 
-    def _delivery(self):
+    # Builds a single delivery for each vendor, aggregating the line items that vendor sold
+    def _deliveries(self):
+        deliveries = []
+
+        for vendor_id in self.by_vendor.keys():
+            line_items = self.by_vendor[vendor_id]
```

Comment:
Sorry didn't realize till now. method looks great.
Just maybe here to avoid self.by_vendor[vendor_id] you can use .items() instead of .keys()
to get the tuple of vendor_id,line_items and save one line. Small detail.

Reply:
Nice! thanks for the heads-up :ok_hand:

Reply:
addressed in https://github.com/coopdevs/lazona_connector/pull/45/commits/e7692f39bffaec44f97eed2c7634a1fa8f717a59